pkg/isb/stores/jetstream/reader.go

```diff
@@ -119,12 +119,27 @@ func (jr *jetStreamReader) Read(_ context.Context, count int64) ([]*isb.ReadMess
 	}(time.Now())
 	var err error
 	var result []*isb.ReadMessage
-	msgs, err := jr.sub.Fetch(int(count), nats.MaxWait(jr.opts.readTimeOut))
+	// We use FetchBatch to have batching in our code with two parameters, ie ReadBatchSize
+	// and ReadTimeout. This function should read from Jetstream until we have a read ReadBatchSize
+	// number of messages or the ReadTimeout duration has elapsed.
+	msgs, err := jr.sub.FetchBatch(int(count), nats.MaxWait(jr.opts.readTimeOut))
 	if err != nil && !errors.Is(err, nats.ErrTimeout) {
 		isbReadErrors.With(map[string]string{"buffer": jr.GetName()}).Inc()
 		return nil, fmt.Errorf("failed to fetch messages from jet stream subject %q, %w", jr.subject, err)
 	}
-	for _, msg := range msgs {
+
+	// Wait until the batch processing is done
+	<-msgs.Done()
+
+	// Check if there was any error while fetching the messages
+	// No error is returned from the FetchBatch method in case of pull request expiry (even if there are no messages).
+	if err = msgs.Error(); err != nil {
+		return nil, err
+	}
```

Comment:
we cannot do this; we will have to return partial read messages, or else it is a data loss scenario. Perhaps you can return err at the end.

Reply:
I can return the partial messages read along with the error.

Though one question, in case we return no messages, they won't be acked by us as well right? And ideally the whole batch should be retried in the next read?